diff --git a/out/macPackager.js b/out/macPackager.js
index 30787ba358acb9277a63df05c839a0caeea8ad91..aa131c6397d274486117f7693d409729ba4d3ca1 100644
--- a/out/macPackager.js
+++ b/out/macPackager.js
@@ -276,7 +276,7 @@ class MacPackager extends platformPackager_1.PlatformPackager {
             await this.doFlat(appPath, artifactPath, masInstallerIdentity, keychainFile);
             await this.dispatchArtifactCreated(artifactPath, null, builder_util_1.Arch.x64, this.computeSafeArtifactName(artifactName, "pkg", arch, true, this.platformSpecificBuildOptions.defaultArch));
         }
-        await this.notarizeIfProvided(appPath);
+        if (!isMas) await this.notarizeIfProvided(appPath);
         return true;
     }
     async getOptionsForFile(appPath, isMas, customSignOptions) {
